src/test/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticleTest.java

```diff
@@ -0,0 +1,189 @@
+import info.freelibrary.util.LoggerFactory;
+
+import edu.ucla.library.iiif.fester.Config;
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import io.vertx.config.ConfigRetriever;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.RunTestOnContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class S3BucketVerticleTest extends AbstractFesterVerticle {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3BucketVerticleTest.class, MESSAGES);
+
+    private static final String MANIFEST_PATH = "src/test/resources/testManifest.json";
+
+    private static final String VERTICLE_NAME = S3BucketVerticle.class.getName();
+
+    private static final String DEFAULT_ACCESS_KEY = "YOUR_ACCESS_KEY";
+
+    private static String s3Bucket = "unconfigured";
+
+    private static AWSCredentials myAWSCredentials;
+
+    @Rule
+    public RunTestOnContext myRunTestOnContextRule = new RunTestOnContext();
+
+    private String myManifestKey;
+
+    /** We can't, as of yet, execute these tests without a non-default S3 configuration */
+    private boolean isExecutable;
+
+    private AmazonS3 myAmazonS3;
+
+    /**
+     * Set up the testing environment.
+     *
+     * @param aContext A test context
+     * @throws Exception If there is trouble starting Vert.x or configuring the tests
+     */
+    @SuppressWarnings("deprecation")
+    @Before
+    public void setUp(final TestContext aContext) throws Exception {
+        final Vertx vertx = myRunTestOnContextRule.vertx();
+        final ConfigRetriever configRetriever = ConfigRetriever.create(vertx);
+        final DeploymentOptions options = new DeploymentOptions();
+        final Async asyncTask = aContext.async();
+
+        configRetriever.getConfig(getConfig -> {
+            if (getConfig.succeeded()) {
+                final JsonObject config = getConfig.result();
+
+                myManifestKey = UUID.randomUUID().toString() + Constants.JSON_EXT;
+
+                // We need to determine if we'll be able to run the S3 integration tests so we can skip if needed
+                if (config.containsKey(Config.S3_ACCESS_KEY) && !config.getString(Config.S3_ACCESS_KEY,
+                        DEFAULT_ACCESS_KEY).equalsIgnoreCase(DEFAULT_ACCESS_KEY)) {
+                    isExecutable = true;
+                }
+
+                vertx.deployVerticle(VERTICLE_NAME, options.setConfig(config), deployment -> {
+                    if (deployment.failed()) {
+                        final Throwable details = deployment.cause();
+                        final String message = details.getMessage();
+
+                        LOGGER.error(details, message);
+                        aContext.fail(message);
+                    }
```

Comment:
Know this is how it is in bucketeer, but it would be good to put the non-`if` stuff here in an `else` since we don't need it to run if the deployment has failed.